src/service/dashboards/mod.rs

```diff
@@ -38,12 +38,36 @@ pub async fn create_dashboard(
     org_id: &str,
     folder_id: &str,
     body: web::Bytes,
-) -> Result<HttpResponse, io::Error> {
+) -> Result<HttpResponse, anyhow::Error> {
     // NOTE: Overwrite whatever `dashboard_id` the client has sent us
     // If folder is default folder & doesn't exist then create it
 
-    match db::folders::get(org_id, folder_id).await {
-        Ok(_) => {
+    if db::folders::exists(org_id, folder_id).await? {
```

Comment:
Sadly, git did not do a great job rendering this diff and it looks like more changes than it really his.

The main idea of the change here is that rather than checking that `db::folders::get(...).await` returns an `Ok` response to determine if the folder exists, I instead check the new function `db::folders.exists(...).await?` to determine if the folder exists or not.

I think the previous logic may have had a subtle issue in that is assumed an `Err` response from `db::folders::get` indicates that the folder does not exist. This isn't strictly true since that function can return `Err` for a variety of reasons (eg: losing connection to the DB for some reason). Therefore I think that using `?` to fast-return an error from the call to the db function  should make this logic behave more correctly when the call to the db function happens to fail.

I also introduced the `exists` function to make it more clear that the Boolean return value, rather then the `Ok`/`Err` status of the response itself is what communicates the existence of the folder.